### PR TITLE
Synchronized TC with HEAD of master branch of emulation.

### PR DESCRIPTION
**Pull request by aehart**
TC is now in synch with the HEAD of the master branch of the emulation. This mostly involved tweaking bit widths, and everything (C-simulation, pipelining, C/RTL cosimulation) still works.

Also removed TC files in emData/ in favor of keeping them on CERNBox. Those on CERNBox now come from the HEAD of the emulation.

### CODE REVIEW

File: TrackletAlgorithm/TrackletCalculator.cc [hunk 1]
```diff
@@ -28,7 +28,7 @@ void TrackletCalculator_L1L2E(
 #pragma HLS array_partition variable=projout_barrel_2s complete
 #pragma HLS array_partition variable=projout_disk complete
 
-  TC_L1L2E: TrackletCalculator<TC::L1L2, TC::E, BARRELPS, BARRELPS, 2, 1, 13, 0x1FE0, 0x0000, 0x7772000, 0x7777, kMaxProc>(
```
Comment (pwittich) [inline TrackletAlgorithm/TrackletCalculator.cc:31]: these magic numbers make me nervous....

File: TrackletAlgorithm/Constants.h [hunk 2]
```diff
@@ -26,12 +26,12 @@ constexpr int TEBinsBits = 3;
 
 // stub digitization constants
 constexpr double kr = 0.0292969;
-constexpr double kphi = 7.8283e-06;
+constexpr double kphi = 7.71297e-06;
 constexpr double kz = 0.0585938;
 
 // tracklet digitization constants
-constexpr double krinv = 1.04377e-06;
```
Comment (pwittich) [inline TrackletAlgorithm/Constants.h:33]: why did these change?

File: TrackletAlgorithm/TrackletCalculator.h [hunk 3]
```diff
@@ -78,18 +79,18 @@ namespace TC {
   static const ap_uint<9> z0cut_L1L2 = 15.0 / kz0 + 1.0e-1;
   static const ap_uint<9> z0cut = 1.5 * 15.0 / kz0 + 1.0e-1;
   static const Types::rmean RMEAN[6] = {
-    25.137 / kr + 1.0e-1,
```
Comment (pwittich) [inline TrackletAlgorithm/TrackletCalculator.h:81]: do these belong in the constants file?

Comment (aehart) [reply TrackletAlgorithm/TrackletCalculator.h:81]: Yes, I think so. I've moved them to Constants.h.

File: TrackletAlgorithm/Constants.h (hunk 2 above)
Comment (aehart) [reply TrackletAlgorithm/Constants.h:33]: I don't actually know. This changed in the emulation, so maybe @djcranshaw or @aryd know?

File: TrackletAlgorithm/TrackletCalculator.cc (hunk 1 above)
Comment (aehart) [reply TrackletAlgorithm/TrackletCalculator.cc:31]: These numbers come directly from the wiring, and I think ultimately, we want a script to generate these top functions automatically from the wires file. I can start working on that, but beyond that, I'm not sure what to do about these numbers.

Comment (pwittich) [reply TrackletAlgorithm/TrackletCalculator.cc:31]: should they at least become named constants so people have a hope of understanding them?

Comment (aehart) [reply TrackletAlgorithm/TrackletCalculator.cc:31]: OK, I moved the magic numbers to a new file called `TrackletCalculator_parameters.h`, which, in the future, can be generated by a script from the wiring file. I decided to organize the numbers into constexpr template functions to reflect the fact that they are functions of the seed and iTC letter.

I also realized that the masks didn't need to be parameters of the top template function anymore, so I removed them.

**Pull request merged**